### PR TITLE
Adds steal research levels objectives to traitors+lings, makes it work for everyone not just ninjas.

### PR DESCRIPTION
**Pull request by silicons**
Steal objectives by using tech disks.
:cl:
rscadd: The syndicate have recently begun sending agents to extract vital research information from Nanotrasen.
/:cl:
I'd like to make this specific levels but I'll have to rethink this when techwebs is done(tm) anyways so this is just raw "how many levels" for now.

### CODE REVIEW

**Comment by Kerbin-Fiber** [comment]
Wow, we should add this in the game!

**Comment by GunHog** [comment]
>Kev creating more work for himself

**Comment by InsaneHyena** [comment]
Can you make it delete research levels, so it would actually be antagonistic, and people would have an incentive to stop it?

**Comment by kevinz000** [comment]
until techwebs is out that's a joke because you can just restore a backup but you can feel free to bug me after i finish techwebs sometime.

**Comment by praisenarsie** [comment]
easiest objective like ever??????????????????????????

**Comment by DoctoreMad** [comment]
On the up side we might get R&D done like 50% of the time instead of the rate it is at now.

File: code/game/gamemodes/objective.dm [hunk 1]
```diff
@@ -553,33 +553,29 @@ GLOBAL_LIST_EMPTY(possible_items_special)
...
-
+/datum/objective/download/check_completion()
+	var/list/current_tech = list()
+	if(ismob(owner.current))
+		var/mob/M = owner.current			//Yeah if you get morphed and you eat a quantum tech disk with the RD's latest backup good on you soldier.
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(H && (H.stat != DEAD) && istype(H.wear_suit, /obj/item/clothing/suit/space/space_ninja))
+				var/obj/item/clothing/suit/space/space_ninja/S = H.wear_suit
+				for(var/datum/tech/T in S.stored_research)
```
Comment (Cyberboss) [inline code/game/gamemodes/objective.dm]: you didn't check if the suit was initialized

**Review by Cyberboss — DISMISSED** [review]

**Comment by TheNightingale** [comment]
Maybe a good idea to not have this objective assigned to science staff. It's like giving the CE "steal the station blueprints".

**Comment by ExcessiveUseOfCobblestone** [comment]
> RND PR
>> not techwebs

**Comment by pubby** [comment]
This being 30% of steal objectives seems too high.

**Comment by Robustin** [comment]
I like this, but only if it includes some negative consequence for RND.

You could even go as far as to require that the station not possess the research levels you stole by the round's end. 

Otherwise its literally an objective that people will give you willingly.

**Comment by Anonmare** [comment]
You could give traitors with that objective a device they have to use on the R&D server console that wipes out every level and every design from the servers its hooked up to.

Traitors could also improvise using tech disks if they somehow lose it

File: code/game/gamemodes/changeling/changeling.dm [hunk 2]
```diff
@@ -131,10 +131,16 @@ GLOBAL_LIST_INIT(slot2type, list("head" = /obj/item/clothing/head/changeling, "w
...
-		var/datum/objective/steal/steal_objective = new
-		steal_objective.owner = changeling
-		steal_objective.find_target()
-		changeling.objectives += steal_objective
+		if(prob(70))
+			var/datum/objective/steal/steal_objective = new
+			steal_objective.owner = changeling
+			steal_objective.find_target()
+			changeling.objectives += steal_objective
+		else
```
Comment (pubby) [inline code/game/gamemodes/changeling/changeling.dm]: Add a check to prevent players from receiving multiple download objectives.

**Comment by YakumoChen** [comment]
@Robustin we already have objectives like Unused Slime Core that are entirely easy to just walk in and steal or even just ask / trade for. Forcing it to be something where any sort of bargaining means fucking over the station entirely (RnD levels are really useful) for the whole round would mean you're pretty much forced to murderbone or gain the wrath of most of the crew.

File: code/game/gamemodes/objective.dm (hunk 1 above)
Comment (kevinz000) [reply code/game/gamemodes/objective.dm]: intended, didn't really make sense to me that you had to have your suit turned on to steal things when it gets stored in the suit either way.

File: code/game/gamemodes/changeling/changeling.dm (hunk 2 above)
Comment (kevinz000) [reply code/game/gamemodes/changeling/changeling.dm]: this proc assigns all objectives at once it doesn't matter.

**Comment by kevinz000** [comment]
lowered chance to 15% instead of steal objectives

File: code/game/gamemodes/changeling/changeling.dm (hunk 2 above)
Comment (pubby) [reply code/game/gamemodes/changeling/changeling.dm]: oops I clicked on the wrong code. I meant in traitor `forge_single_objective`

**Comment by BeeSting12** [comment]
@YakumoChen 
I think a supermatter sliver starts a meltdown

**Comment by kevinz000** [comment]
@BeeSting12 if you suck it does it's just 200 matter power that's manageable.

**Comment by kevinz000** [comment]
@GunHog this exact thing is coded for techwebs on the branch I'm just making it work earlier for feedback

**Comment by DoctoreMad** [comment]
>we already have objectives like Unused Slime Core that are entirely easy to just walk in and steal or even just ask / trade for. 

Wait are there people that support those?

**Comment by ExcessiveUseOfCobblestone** [comment]
>@Robustin we already have objectives like Unused Slime Core that are entirely easy to just walk in and steal or even just ask / trade for. Forcing it to be something where any sort of bargaining means fucking over the station entirely (RnD levels are really useful) for the whole round would mean you're pretty much forced to murderbone or gain the wrath of most of the crew.

Not to mention figuring a way to deny the crew RnD levels is next to impossible anyway.

I can't think of a single objective item that would completely cripple the crew if taken. A few are really, really useful to have (hand tele in particular) but nothing that is so outrageously useful that not having it might realistically force a shuttle call or spark a manhunt over.

It would be a lot better if steal objectives involved stealing things that are imperative to departments or unique to force conflicts and rough decisions.

They _are_ called steal objectives for a reason, you aren't meant to be handed the item and tada win.

**Comment by PKPenguin321** [comment]
Nobody does research and you get this objective and you're not a powergamer who memorized the R&D flowchart

What do?

**Comment by Bawhoppen** [comment]
@PKPenguin321 we can add a list of known powergamers and check against it if any are in the round and only give it if they are

**Comment by kevinz000** [comment]
@PKPenguin321 techwebs will solve that anyhow but it's not an issue with current rnd either.

max levels: 20
that's about 3 in each category if you use all but illegal/alien.
anyone can get that by just printing thigns and stuffing them in the DA, and stuff like that.

File: code/game/gamemodes/changeling/changeling.dm (hunk 2 above)
Comment (kevinz000) [reply code/game/gamemodes/changeling/changeling.dm]: fixed

**Pull request merged**